libraries/AP_Vehicle/AP_Vehicle.cpp

```diff
@@ -213,6 +214,14 @@ void AP_Vehicle::send_watchdog_reset_statustext()
         );
 }
 
+bool AP_Vehicle::is_crashed() const
+{
+    if (AP::arming().is_armed()) {
```

Comment:
So you're keen to stick with using the AP::arming().is_armed() instead of hal's soft armed?